### PR TITLE
chore: add EVM extensions docs

### PR DESCRIPTION
**Pull request by GAtom22**
## Description

Add EVM extensions documentation included on v13

Closes https://linear.app/evmos/issue/ENG-1310/precompile-docs-for-developers

### CODE REVIEW

**Review by fedekunze — APPROVED** [review]
ACK

File: docs/develop/build-a-dApp/build-smart-contracts/evm_extensions.md [hunk 1]
```diff
@@ -8,17 +8,20 @@ sidebar_position: 3
...
+| `0x0000000000000000000000000000000000000003` | RIPEMD-160 Hash     | No       |                                                   |
+| `0x0000000000000000000000000000000000000004` | Data Copy           | No       |                                                   |
+| `0x0000000000000000000000000000000000000005` | ExpMod              | No       | [EIP-198](https://eips.ethereum.org/EIPS/eip-198) |
+| `0x0000000000000000000000000000000000000006` | BN256Add            | No       | [EIP-196](https://eips.ethereum.org/EIPS/eip-196) |
+| `0x0000000000000000000000000000000000000007` | BN256ScalarMul      | No       | [EIP-196](https://eips.ethereum.org/EIPS/eip-196) |
+| `0x0000000000000000000000000000000000000008` | BN256Pairing        | No       | [EIP-197](https://eips.ethereum.org/EIPS/eip-197) |
+| `0x0000000000000000000000000000000000000009` | Blake2F             | No       | [EIP-152](https://eips.ethereum.org/EIPS/eip-152) |
+| `0x0000000000000000000000000000000000000010` | Bech32 encoding     | No       |                                                   |
```
Comment (fedekunze) [inline docs/develop/build-a-dApp/build-smart-contracts/evm_extensions.md:22]: actually, this address is wrong. See the Notion doc with the guidelines

Comment (GAtom22) [reply docs/develop/build-a-dApp/build-smart-contracts/evm_extensions.md:22]: @fedekunze is like that on the extensions repo. Could you please indicate which is the notion doc that you're referring to?

File: docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/staking.md [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+## Solidity Interface & ABI
+
+`Staking.sol` is an interface through which Solidity contracts can interact with Cosmos SDK staking.
+This is convenient for developers as they don’t need to know the implementation details behind the `x/staking`
+module in the Cosmos SDK. Instead,
+they can interact with staking functions using the Ethereum interface they are familiar with.
+
+### Interface `Staking.sol`
+
+```solidity
```
Comment (fedekunze) [inline docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/staking.md:16]: I'd just link to the file on the extensions repo

File: docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/authorization.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+This is achieved by the `Authorization.sol` and `GenericAuthorization.sol`
+that provide the necessary functions to grant approvals and allowances.
+The precompiled contracts use these interfaces, `AuthorizationI` and `GenericAuthorizationI`,
+to allow users to approve the corresponding messages and amounts if needed.
+
+## Solidity Interfaces
+
+### `Authorization.sol`
+
+```solidity
```
Comment (fedekunze) [inline docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/authorization.md:18]: link to file on extensions repo

File: docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/distribution.md [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+## Solidity Interface & ABI
+
+`Distribution.sol` is an interface through which Solidity contracts can interact with Cosmos SDK distribution.
+This is convenient for developers as they don’t need to know the implementation details behind
+the `x/distribution` module in the Cosmos SDK. Instead,
+they can interact with staking functions using the Ethereum interface they are familiar with.
+
+### Interface `Distribution.sol`
+
+```solidity
```
Comment (fedekunze) [inline docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/distribution.md:16]: ditto

File: docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/distribution.md [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+    event WithdrawValidatorCommission(
+        string indexed validatorAddress,
+        uint256 commission
+    );
+}
+```
+
+### ABI
+
+The minified version of the precompiled contract `abi.json`:
```
Comment (fedekunze) [inline docs/develop/build-a-dApp/build-smart-contracts/evm-extensions/distribution.md:163]: ditto link

**Review by fedekunze — APPROVED** [review]
ACK minor suggestions

**Pull request merged**